backend/src/game/PokerTable.ts

```diff
@@ -1,103 +1,92 @@
+      seatsArray.push(newSeat);
+      seatNumber++;
     }
+    this.seats = seatsArray;
+  }
 
-    public static createPokerTable(tableName: string, numberOfSeats: number): Result<PokerTable>{
-        const res = Rooms.createRoom(tableName);
-        if (res.ok){
-            const roomId = res.getValue();
-            const newTable = new PokerTable(tableName, numberOfSeats, roomId);
-
-            // Store the new table in the in-memory database
-            PokerTable.tables[roomId] = newTable;
+  public static createPokerTable(tableName: string, numberOfSeats: number): Result<PokerTable> {
+    const res = Rooms.createRoom(tableName);
+    if (res.ok) {
+      const roomId = res.getValue();
+      const newTable = new PokerTable(tableName, numberOfSeats, roomId);
 
-            return new ResultSuccess(newTable);
-        }
-        return new ResultError(res.errorMessage);
+      return new ResultSuccess(newTable);
     }
+    return new ResultError(res.errorMessage);
+  }
 
-    public static sitAtTable(tableName: string, seatNumber: string, clientId: string): Result<void> {
-        let table = this.getTable(tableName);
-        for (const seat of table.seats) {
-            // Check if player is already seated at the table
-            if (seat.playerId == clientId ){
-                return Result.error('Player is already sitting at the table');
-            }
-        }
-        for (const seat of table.seats) {
-            // Check if the seat is already taken by someone else
-            if (seat.seatNumber == seatNumber) {
-                if (seat.isTaken){
-                    return Result.error('Seat is already taken');
-                } else {
-                    // Update the seat properties
-                    seat.playerId = clientId;
-                    seat.isTaken = true;
-                    // check if table is fully/ready
-                    const tableIsReady = this.checkTableReady(table)
-                    // Emit event to all clients connected that game is starting
-                    if (tableIsReady) {
-                        let event = 'start_game';
-                        let payload = {
-                            tableName: tableName
-                        };
-                        let send_events = Rooms.sendEventToRoom('table_1', event, payload);
-                        if (!send_events.ok) {
-                            return Result.error(send_events.errorMessage);
-                        }
-                    }
-                    return Result.success();
-                }
+  public sitAtTable(tableName: string, seatNumber: string, clientId: string): Result<void> {
+    for (const seat of this.seats) {
+      // Check if player is already seated at the table
+      if (seat.playerId == clientId) {
+        return Result.error('Player is already sitting at the table');
+      }
+    }
+    for (const seat of this.seats) {
```

Comment:
Cool that now it's an instance of a PokerTable we can use the this. context. So clean.

Reply:
Yeah it now reads correctly now eh?